classes/teacher.rb

```diff
@@ -1,6 +1,6 @@
 class Teacher < Person
-  def initialize(specialization, age, name)
-    super(age, name)
+  def initialize(specialization, age, name, parent_permission)
+    super(age, name, parent_permission: parent_permission)
```

Comment:
- [ ] Kindly make sure to implement the changes asked by the previous reviewer. Follow below:
```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```